onnxruntime/core/providers/mkldnn/activation/activations.cc

```diff
@@ -0,0 +1,238 @@
+/* Copyright(C) 2018 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions
+and limitations under the License.
+==============================================================================*/
+
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "core/providers/mkldnn/mkldnn_common.h"
+#include "core/providers/mkldnn/activation/activations.h"
+#include "core/providers/mkldnn/mkldnn_fwd.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+namespace {
+// Struct which encapsulates parameters for MKLDNN Pool primitive.
+struct ReluParams {
+  mkldnn::memory::dims& src_dims;
+  mkldnn::memory::dims& dst_dims;
+  size_t num_dimensions;
+
+  ReluParams(mkldnn::memory::dims& src_dims, mkldnn::memory::dims& dst_dims, 
+		size_t dimensions = 0)
+      : src_dims(src_dims),
+        dst_dims(dst_dims),
+        num_dimensions(dimensions) {}
+
+  // Used as the key for Pool Primitive Reuse Pool.
+  std::string ToString() const {
+    std::string key;
+    key.reserve(128);
+    key.append("Relu_");
+    AddDimsToKey(key, src_dims);
+    AddDimsToKey(key, dst_dims);
+    return key;
+  }
+};
+
+template <typename T>
+class ReluPrimitive final : public PrimitiveBase {
+ public:
+  explicit ReluPrimitive(const ReluParams& params)
+      : cpu_engine_(GetEngine()) {
+    context_.stream.reset(new mkldnn::stream(mkldnn::stream::kind::eager));
+    if (context_.relu_fwd == nullptr) {
+      Initialize(params);
+    }
+  }
+
+  ~ReluPrimitive() = default;
+
+  void Compute(const T* src_data, const T* dst_data) {
+	  context_.src_mem->set_data_handle(
+		  static_cast<void*>(const_cast<T*>(src_data)));
+	  context_.dst_mem->set_data_handle(
+		  static_cast<void*>(const_cast<T*>(dst_data)));
+	  context_.stream->submit(context_.net);
+
+	  context_.src_mem->set_data_handle(nullptr);
+	  context_.dst_mem->set_data_handle(nullptr);
+	  return;
+  }
+
+  mkldnn::memory::format GetSrcMemoryFormat() const { return context_.src_fmt; }
+  mkldnn::memory::format GetDstMemoryFormat() const { return context_.dst_fmt; }
+
+  std::unique_ptr<mkldnn::memory::desc> 
+	GetDstMemoryDesc() const { return context_.dst_md; }
+
+  std::unique_ptr<mkldnn::eltwise_forward::primitive_desc> 
+	GetPrimitiveDesc() const {
+		return context_.relu_fwd_pd;
+  }
+
+ private:
+  struct ReluContext {
+    mkldnn::memory::format src_fmt;
```

Comment:
per comment above. remove src_fmt, and dst_fmt if no need for them.

Reply:
removed